SteamProject/MySteam/Steam.cpp

```diff
@@ -155,87 +143,133 @@ void FSteam::OpenCategoriesMenu()
-		{
-			if (MainHandler.GetCategoriesAmount() == 0)
-			{
-				std::cout << "There are no categories saved." << std::endl;
-				std::cin.ignore();
-				std::cin.get();
-			}
-			else
-			{
-				do
-				{
-					system("cls");
-					MainHandler.ShowCategories();
-					int SelectedCategoryIndex = ValidateInput("Choose a category");
-					if (SelectedCategoryIndex >= 0 && SelectedCategoryIndex < MainHandler.GetCategoriesAmount())
-					{
-						MainHandler.DeleteCategory(SelectedCategoryIndex);
-						break;
-					}
-				} while (true);
-			}
-		}
-		else
-		{
-			CategoriesOption = 0;
-		}
+		case 1:
+			AddCategoryMenu();
+			break;
 
+		case 2:
+			DeleteCategoryMenu();
+			break;
+		}
 
-	} while (CategoriesOption == 0);
+	} while (CategoriesOption < 1 || CategoriesOption > 2);
 }
 
+//-------------------------------------------------------------------------------------------------------------
+
 void FSteam::OpenShowGamesMenu()
 {
 	system("cls");
 	std::cout << "\n---------------------------------------------------" << std::endl;
 	std::cout << "\tCategory name: " << Uncategorized.GetCategoryName() << std::endl;
 	Uncategorized.ShowAllGames();
 
+	std::cout << "Games: " << std::endl;
 	MainHandler.ShowAllCategories();
 
 	std::cout << "\npress ENTER to return to main menu...";
 	std::cin.ignore();
 	std::cin.get();
 }
+
+//-------------------------------------------------------------------------------------------------------------
+
+bool FSteam::AddCategoryMenu()
+{
+	if (MainHandler.HasSpace())
+	{
+		system("cls");
+		std::cin.ignore(100, '\n');
+		std::cout << "Enter the name of the category >> ";
+		std::string NewCategoryName;
+		std::getline(std::cin, NewCategoryName);
+
+		if (MainHandler.AddCategory(NewCategoryName))
+		{
+			std::cout << "Category added!" << std::endl;
+			std::cin.ignore();
+			//std::cin.get();
+			return true;
+		}
+		else
+		{
+			std::cout << "Invalid name!" << std::endl;
+			std::cin.ignore();
+			std::cin.get();
+			return false;
+		}
+	}
+	else
+	{
+		std::cout << "You don't have more space for new categories!" << std::endl;
+		std::cin.ignore();
+		std::cin.get();
+		return false;
+
+	}
+}
+
+//-------------------------------------------------------------------------------------------------------------
+
+void FSteam::DeleteCategoryMenu()
+{
+	if (MainHandler.GetCategoriesAmount() == 0)
+	{
+		std::cout << "There are no categories saved." << std::endl;
+		std::cin.ignore();
+		std::cin.get();
+	}
+	else
+	{
+		do
+		{
+			system("cls");
+			MainHandler.ShowCategories();
+			int SelectedCategoryIndex = ValidateInput("Choose a category");
+			if (SelectedCategoryIndex >= 0 && SelectedCategoryIndex < MainHandler.GetCategoriesAmount())
+			{
+				MainHandler.DeleteCategory(SelectedCategoryIndex);
+				break;
+			}
+		} while (true);
+	}
+}
+
+//-------------------------------------------------------------------------------------------------------------
+
+bool FSteam::AddGameMenu(const FGame& InGame)
```

Comment:
The name of this function tells me that you are adding a new game menu, but you are passing in a game, so unless I read inside the function, I won't understand what you mean. Do you mean something like `AddGame`?

Same thing for AddCategoryMenu and DeleteCategoryMenu.